runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -545,14 +566,16 @@ public void clear() {
     @Override
     @Accessor
     public @Nonnull Set<K> keySet() {
-        return ImmutableSet.copyOf(mainMap.keySet());
+        return mainMap.entryStream().map(Entry::getKey)
```

Comment:
Mostly idiomatic. All bulk-level APIs have to consume the data via entryStream(), since it is not guaranteed that keySet() valueSet() and entrySet() are actually implemented.